controllers/vspherecluster_controller.go

```diff
@@ -334,7 +366,60 @@ func cloudProviderConfigurationAvailable(ctx *context.ClusterContext) bool {
 	return !reflect.DeepEqual(ctx.VSphereCluster.Spec.CloudProviderConfiguration, infrav1.CPIConfig{})
 }
 
+func (r clusterReconciler) reconcileIdentitySecret(ctx *context.ClusterContext) error {
+	vsphereCluster := ctx.VSphereCluster
+	if identity.IsSecretIdentity(vsphereCluster) {
+		secret := &apiv1.Secret{}
+		secretKey := client.ObjectKey{
+			Namespace: vsphereCluster.Namespace,
+			Name:      vsphereCluster.Spec.IdentityRef.Name,
+		}
+		err := ctx.Client.Get(ctx, secretKey, secret)
+		if err != nil {
+			return err
+		}
+
+		// check if cluster is already an owner
+		if !clusterutilv1.IsOwnedByObject(secret, vsphereCluster) {
+			secretMU.Lock()
```

Comment:
we don't need the mutex here too, we operate against a local copy, and optimistic concurrency would trigger a conflict